src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySourceConfig.java

```diff
@@ -203,6 +214,17 @@ private void validateTable(FailureCollector collector) {
     }
   }
 
+  private void validatePartitionDate(FailureCollector collector, String partitionDate, String fieldName) {
+    SimpleDateFormat date = new SimpleDateFormat(VALID_DATE_FORMAT);
+    date.setLenient(false);
+    try {
+      date.parse(partitionDate);
+    } catch (ParseException e) {
+      collector.addFailure(String.format("%s is not in a valid format for %s.", partitionDate, fieldName),
+                           String.format("Enter valid date in format: %s", VALID_DATE_FORMAT));
```

Comment:
you can pass in NAME_PARTITION_FROM/NAME_PARTITION_TO, specify .withConfigProperty() on the failure collector so that the field that was configured incorrectly gets highlighted

Reply:
+1, Thanks for catching that.